pkgs/by-name/mi/minipom/package.nix

```diff
@@ -0,0 +1,101 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  cargo-tauri,
+  darwin,
+  glib-networking,
+  nodejs_22,
+  openssl,
+  pkg-config,
+  wrapGAppsHook3,
+  gobject-introspection,
+  at-spi2-atk,
+  atkmm,
+  cairo,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  harfbuzz,
+  librsvg,
+  libsoup_3,
+  pango,
+  webkitgtk_4_1,
+  alsa-lib,
+  zlib,
+  pnpm_9,
+  fetchFromGitHub,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "minipom";
+  version = "0.9.3";
+
+  src = fetchFromGitHub {
+    owner = "tfkhdyt";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-hUhGU2w4ndmU7KpFr+E4tTWkvpI8PuglPXRkyf1Q2Ys=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
```

Comment:
How can i do that? It's my first time publishing Rust Tauri app

Reply:
You add `cargoHash = ""` and rebuild, then put in the correct hash.

Reply:
Seems like imported cargoLock is required by `cargo-tauri.hook`, I get this error after replacing cargoLock.lockFile to cargoHash:

```
Running phase: unpackPhase
unpacking source archive /nix/store/q4q0yid5kwz64lvfl80ixgbdhymmq43c-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase

ERROR: The Cargo.lock file doesn't exist

Cargo.lock is needed to make sure that cargoHash/cargoSha256 doesn't change
when the registry is updated.
```

This is another Tauri app in nixpkgs that does the same thing regarding importing Cargo.lock:

https://github.com/NixOS/nixpkgs/tree/6af89f26532b8255bd5cecba35a1ebf72da3892b/pkgs/by-name/po/pot

Reply:
The error is somewhat misleading. There is a `Cargo.lock` in the src-tauri directory.

Reply:
Try `sourceRoot = "${finalAttrs.src.name}/src-tauri";`

Reply:
> This is another Tauri app in nixpkgs that does the same thing regarding importing Cargo.lock:

That is because it hasn't been updated to the new fetcher, I believe.

Reply:
Done, it's working now. Thank you